analyses/fusion-summary/01-fusion-summary.Rmd

```diff
@@ -203,9 +257,104 @@ allFuseEwing %>%
   write_tsv(ewingsFile)
 ```
 
+#### Perform selection for LGAT fusions
+
+```{r}
+# Which genes/fusions are not kinases, but in the list?
+# Separate LGAT fusions into genes, combine with gene list, check for not kinase
+lgatFuses_df <- as.data.frame(lgatFuses) %>%
+  separate(lgatFuses, into = c("Gene1A", "Gene1B"), remove = F)
+kinases <- annot %>%
+  filter(type == "Kinase")
+lgatFuses_df$Gene1A_anno <- ifelse(lgatFuses_df$Gene1A %in% kinases$Gene_Symbol, "kinase", "")
+lgatFuses_df$Gene1B_anno <- ifelse(lgatFuses_df$Gene1B %in% kinases$Gene_Symbol, "kinase", "")
+# Only pull fusions that do not contain kinase genes, as ones with kinases will be dealt with sepaately later
+lgatFuses_df <- lgatFuses_df %>%
+  filter(lgatFuses_df$Gene1A_anno != "kinase" & lgatFuses_df$Gene1B_anno != "kinase") %>%
+  select(lgatFuses)
+nonkinase_lgatFuses <- lgatFuses_df$lgatFuses
+
+# Identify non-kinase genes in LGAT goi list
+nonkinase_lgatGenes <- setdiff(lgatGenes, kinases$Gene_Symbol)
+
+# Pull LGAT non-kinase fusions
+nonkinaseLGAT <- filterFusion(df = putativeOncogenicDF,
+                             fuses = nonkinase_lgatFuses,
+                             genes = nonkinase_lgatGenes) %>%
+  distinct()
+
+# First, filter all fusion dataframe for 3' kinases which are in-frame and retain the kinase domain - keep these
+three_prime_kinase_inframe <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1B_anno) & Fusion_Type == "in-frame" & DomainRetainedGene1B == "Yes") %>%
+  select(Sample, FusionName, Gene1A, Gene1B) %>%
+  distinct()
+
+# Which fusions are not in-frame?
+three_prime_kinase_outframe <- allFuseLGAT %>%
+  filter(grepl("Kinase", Gene1B_anno) & Fusion_Type != "in-frame") %>%
+  distinct()
+# Check they don't retain the kinase domain. They do not, so leave out.
+table(three_prime_kinase_outframe$DomainRetainedGene1B)
```

Comment:
If we have things being printed out, it would be easier if this were at the end of a chunk so we can see the output printed nearer to the code that's responsible for it. 

In general it might be nice to have this one very long chunk broken up into smaller chunks that are more focused.

Reply:
will break these up some